conferences/views.py

```diff
@@ -21,3 +26,29 @@ def index(request):
         }
         context['gapi_place_src'] = GAPI_PLACE_BASE_URL + '?' + urlencode(query)
     return render(request, 'conferences/index.html', context)
+
+
+@login_required
+def register(request, zosia_id):
+    zosia = get_object_or_404(Zosia, pk=zosia_id)
+    ctx = {}
+    form_args = {}
+
+    user_prefs = UserPreferences.objects.filter(zosia=zosia, user=request.user).first()
```

Comment:
jak potrzebujesz informacji nt zosi w środku to możesz zrobić `UserPreferences.objects.select_related('zosia')...`